pkgs/applications/video/pyca/default.nix

```diff
@@ -1,25 +1,70 @@
-{ stdenv, lib, buildPythonApplication, fetchFromGitHub, pycurl, python-dateutil, configobj, sqlalchemy, sdnotify, flask }:
+{ stdenv, lib, buildPythonApplication, python3, fetchFromGitHub, buildNpmPackage }:
 
+let
+  python = python3.override {
+    packageOverrides = self: super: {
+      # pyCA is incompatible with SQLAlchemy 2.0
+      sqlalchemy = super.sqlalchemy.overridePythonAttrs (old: rec {
+        version = "1.4.46";
+        src = self.fetchPypi {
+          pname = "SQLAlchemy";
+          inherit version;
+          hash = "sha256-aRO4JH2KKS74MVFipRkx4rQM6RaB8bbxj2lwRSAMSjA=";
+        };
+      });
+    };
+  };
+
+  frontend = buildNpmPackage rec {
+    pname = "pyca";
+    version = "4.5";
+
+    src = fetchFromGitHub {
+      owner = "opencast";
+      repo = "pyCA";
+      rev = "v${version}";
+      sha256 = "sha256-cTkWkOmgxJZlddqaSYKva2wih4Mvsdrd7LD4NggxKQk=";
+    };
+
+    npmDepsHash = "sha256-0U+semrNWTkNu3uQQkiJKZT1hB0/IfkL84G7/oP8XYY=";
+
+    nativeBuildInputs = [ python ];
+
+    patches = [ ./add-package-json-version.patch ];
```

Comment:
That version needs to be updated whenever we update the npm package version?
Maybe put a placeholder in the patch like `@version` and than use the substitute function in bash to replace it.

Reply:
It's also possible to use `jq` and it's setpath function

Reply:
Uhh, I like the `jq` idea, I'll try that.

The version really is just to shut up whatever `buildNpm` does, it's complaining that a proper package needs a name and a version.